File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.js
// dev/stdin
const fs = require("fs");
let input = fs.readFileSync('dev/stdin').toString().trim().split('\n');

const n = Number(input[0])

let set = new Set()

for(let i = 1; i <= n; i++) {
  const temp = input[i].split(' ')
  if(temp[1].includes('enter')) {
    set.add(temp[0])
  } else {
    set.delete(temp[0])
  }
}
const a = 'a'.toLocaleLowerCase

console.log([...set].sort().reverse().join('\n'))

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys

N = int(sys.stdin.readline().strip())        

answer = ''
company = set()

for _ in range(N):
  name, commute = sys.stdin.readline().strip().split(' ')

  if commute == 'enter':
    company.add(name)
  else:
    company.remove(name)


company = list(company)
company.sort(reverse=True)

for name in company:
  answer += name + '\n'

print(answer)
